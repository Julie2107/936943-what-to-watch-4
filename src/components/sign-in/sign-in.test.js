import React from "react";
import renderer from "react-test-renderer";
import history from "../../history";
import {Router} from "react-router-dom";
import configureStore from 'redux-mock-store';

import SignIn from "./sign-in.jsx";
import NameSpace from "../../reducer/name-space";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

it(`SignIn page is rendered correctly`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
      isAuthError: false
    }
  });
  const tree = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <SignIn
            onSubmit={() => {}}
            authorizationStatus={`AUTH`}
            isAuthError={false}
          />
        </Provider>
      </Router>, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

  expect(tree).toMatchSnapshot();
});
