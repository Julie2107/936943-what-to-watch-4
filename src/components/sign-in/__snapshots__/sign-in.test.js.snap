// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn page is rendered correctly 1`] = `null`;
